doc/fileformat/cable_cell.rst

```diff
@@ -0,0 +1,429 @@
+
+Morphology
+----------
+
+The morphology of a cable cell can be described in terms of points, tagged segments and branches.
+
+.. label:: (point x:real y:real z:real radius:real)
+
+   This describes a 3D *point* in space with ``x``, ``y``, and ``z`` coordinates and a radius ``r`` (unit [µm]).
+
+.. label:: (segment id:int prox:point dist:point tag:int)
+
+   This describes a tapered segment from point ``prox`` to point ``dist`` with a tag ``tag`` and id ``id``.
+   For example:
+
+   .. code:: lisp
+
+      (segment 3 (point 0 0 0 5) (point 0 0 10 2) 1)
+
+   This expression creates a segment with id 3, with a radius that tapers linearly from 5 to 2 µm, which has a
+   a tag of 1.
+
+.. label:: (branch id:int parent_id:int seg:segment [...seg:segment])
+
+   This describes a branch with a given ``id`` which has as a parent the branch with id ``parent_id`` (a
+   ``parent_id`` equal to -1 means the branch is at the root of the morphology). The branch is composed of 1 or
+   more contiguous segments ``seg``.
+
+
+.. label:: (morphology [...b:branch])
+
+   This creates the morphology from a set of branches. There exists more than one valid s-expression to
+   describe the same morphology.
+
+   For example, the shown morphology can be represented using the following s-expression. If we change
+   any of the branch or segment ids, we would obtain an identical morphology.
+
+   .. figure:: ../gen-images/label_morph.svg
+     :width: 600
+     :align: center
+
+     On the left the morphology visualized using its segments, on the right using its branches.
+     Python code to generate this cable cell is in the :class:`segment_tree<arbor.segment_tree>`
+     documentation :ref:`here <morph-label-seg-code>`.
+
+   .. code:: lisp
+
+      (morphology
+        (branch 0 -1
+          (segment 0 (point 0 0 0 2) (point 4 0 0 2) 1)
+          (segment 1 (point 4 0 0 0.8) (point 8 0 0 0.8) 3)
+          (segment 2 (point 8 0 0 0.8) (point 12 -0.5 0 0.8) 3))
+        (branch 1 0
+          (segment 3 (point 12 -0.5 0 0.8) (point 20 4 0 0.4) 3)
+          (segment 4 (point 20 4 0 0.4) (point 26 6 0 0.2) 3))
+        (branch 2 0
+          (segment 5 (point 12 -0.5 0 0.5) (point 19 -3 0 0.5) 3))
+        (branch 3 2
+          (segment 6 (point 19 -3 0 0.5) (point 24 -7 0 0.2) 3))
+        (branch 4 2
+          (segment 7 (point 19 -3 0 0.5) (point 23 -1 0 0.2) 3)
+          (segment 8 (point 23 -1 0 0.3) (point 26 -2 0 0.2) 3))
+        (branch 5 -1
+          (segment 9 (point 0 0 0 2) (point -7 0 0 0.4) 2)
+          (segment 10 (point -7 0 0 0.4) (point -10 0 0 0.4) 2)))
+
+Cable cell
+----------
+
+The entire cable-cell can then be constructed given the 3 previously described component
+expressions.
+
+.. label:: (cable-cell morph:morphology dec:decor dict:label-dict)
+
+   The arguments of the cable-cell can be in any order, as long as all 3 components are listed.
+   For example:
+
+   .. code:: lisp
+
+      (cable-cell
```

Comment:
The indentation does not matter, but is it outputted this way? Also: the placement of linebreaks are a matter of choice/optional. Since few users will be familiar with lispy formats, it might be nice to state these properties explicitly.

Reply:
The output respects the indentation. I'll add a note.

Reply:
:+1: